src/main/java/org/victorrobotics/frc/dtlib/log/DTLogger.java

```diff
@@ -0,0 +1,202 @@
+    return nextHandle++;
+  }
+
+  public static boolean logNewTimestamp() {
+    long newTime = RobotController.getFPGATime() / 1000;
+    long diff = newTime - lastTimestamp;
+    if (diff <= 0) {
+      // Same time, no change
+      return false;
+    }
+    lastTimestamp = newTime;
+
+    if (diff <= 0xFFFF) {
+      // 2 byte form, maximum of 65.535 seconds
+      dataWriter.writeShort(0x0001);
+      dataWriter.writeShort((short) diff);
+    } else {
+      // write new timestamp
+      dataWriter.writeShort(0x0002);
+      dataWriter.writeLong(newTime);
+    }
+    return true;
+  }
+
+  public static boolean init() {
+    if (!isTimeSynchronized()) {
+      return false;
+    }
+
+    try {
+      openCapture();
+      writeHeader();
+      return true;
+    } catch (IOException e) {
+      try {
+        if (dataWriter != null) {
+          dataWriter.close();
+        }
+      } catch (IOException e2) {
+      }
+      dataWriter = null;
+      return false;
+    }
+  }
+
+  private static void openCapture() throws IOException {
+    Instant now = Clock.systemUTC()
+                       .instant();
+    lastTimestamp = RobotController.getFPGATime() / 1_000;
+    startTimeMillis = now.toEpochMilli();
+
+    String name = TIME_FORMATTER.format(now) + ".dtlog";
+    file = new File(LOG_DIRECTORY, name);
+    dataWriter = new DTLogWriter(file, 65536);
+  }
+
+  private static void writeHeader() {
+    dataWriter.writeBytes(DTLIB_LOGGER_MAGIC.getBytes(StandardCharsets.UTF_8));
+
+    dataWriter.writeShort(DTLibInfo.Version.YEAR);
+    dataWriter.writeByte(DTLibInfo.Version.MAJOR);
+    dataWriter.writeByte(DTLibInfo.Version.MINOR);
+
+    String[] wpilibVersion = WPILibVersion.Version.split("\\.");
+    dataWriter.writeShort(Integer.parseInt(wpilibVersion[0]));
+    dataWriter.writeByte(Integer.parseInt(wpilibVersion[1]));
+    dataWriter.writeByte(Integer.parseInt(wpilibVersion[2]));
+
+    dataWriter.writeShort(getTeamNumber());
+    dataWriter.writeShort(0); // unused flags
+    dataWriter.writeLong(startTimeMillis);
+  }
+
+  private static boolean isTimeSynchronized() {
+    return DriverStation.isDSAttached() && LocalDate.now(Clock.systemUTC())
+                                                    .getYear() >= 2000;
+  }
+
+  public static void logDebug(String msg) {
+    logMessage(msg, 0x0008);
+  }
+
+  public static void logInfo(String msg) {
+    logMessage(msg, 0x0009);
+  }
+
+  public static void logWarning(String msg) {
+    logMessage(msg, 0x000a);
+  }
+
+  public static void logError(String msg) {
+    logMessage(msg, 0x000b);
+  }
+
+  private static void logMessage(String msg, int type) {
+    dataWriter.writeShort(type);
+    dataWriter.writeStringUTF8(msg);
+  }
+
+  public static void flush() {
+    dataWriter.flush();
+  }
+
+  private static int getTeamNumber() {
+    if (RobotBase.isSimulation()) {
+      // Can't use IP address in a simulation, return unknown
+      return 0xffff;
```

Comment:
getTeamNumber returns -1 in this case, two other places us 0xffff, seems like a named constant might help.